src/opm/parser/eclipse/Parser/ErrorGuard.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2019 Joakim Hove/datagr
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <iostream>
+#include <opm/parser/eclipse/Parser/ErrorGuard.hpp>
+
+
+namespace Opm {
+
+    void ErrorGuard::add_error(const std::string& errorKey, const std::string& msg) {
+        this->error_list.emplace_back(errorKey, msg);
+    }
+
+
+    void ErrorGuard::add_warning(const std::string& errorKey, const std::string& msg) {
+        this->warning_list.emplace_back(errorKey, msg);
+    }
+
+
+    void ErrorGuard::dump() const {
+        if (!this->warning_list.empty()) {
+            std::cerr << "Warnings:" << std::endl;
+            for (const auto& pair : this->warning_list)
+                std::cerr << pair.first << " : " << pair.second << std::endl;
+        }
+
+
+        if (!this->error_list.empty()) {
+            std::cerr << std::endl << std::endl << "Errors:" << std::endl;
+            for (const auto& pair : this->error_list)
+                std::cerr << pair.first << " : " << pair.second << std::endl;
+        }
+    }
+
+
+    ErrorGuard::~ErrorGuard() {
+        if (*this) {
```

Comment:
Should not the destructor dump() unconditionally? This way, the warnings only get dumped if there are also errors.

Reply:
As already mentioned the warnings have already been passed to the `OpmLog` the normal way. My thought is that *iff* there are errors it might also be interesting to see warnings in the same screenfull - the root cause of the problem might be in the warnings?

If there are only warnings the `ErrorGuard` class should finish and go out of scope silently.